specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingProfile.json

```diff
@@ -69,327 +82,540 @@
+          "type": "array",
           "items": {
-            "$ref": "#/definitions/TargetCloud"
+            "description": "Identifies the cloud environments that are associated with an invoice section. This is a system managed optional field and gets updated as the invoice section gets associated with accounts in various clouds.",
+            "enum": [
+              "Unknown",
+              "AirgapTest",
+              "USGov",
+              "USNat",
+              "USSec"
+            ],
+            "type": "string",
+            "x-ms-enum": {
+              "name": "TargetCloud",
+              "modelAsString": true
+            }
           },
           "readOnly": true,
-          "type": "array"
+          "x-ms-identifiers": []
+        }
+      }
+    },
+    "DeleteBillingProfileEligibilityDetail": {
+      "description": "Validation details of delete billing profile eligibility.",
+      "type": "object",
+      "properties": {
+        "code": {
+          "description": "Code of the delete invoice section eligibility response.",
+          "enum": [
+            "None",
+            "AzureSubscriptions",
+            "ActiveAzureCredits",
+            "ActiveCreditCard",
+            "LastBillingProfile",
+            "NotSupported",
+            "OutstandingCharges",
+            "RecurringCharges",
+            "PendingCharges",
+            "ReservedInstances",
+            "ActiveBillingSubscriptions"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "DeleteBillingProfileEligibilityCode",
+            "modelAsString": true
+          }
+        },
+        "message": {
+          "description": "Validation message.",
+          "type": "string"
+        }
+      }
+    },
+    "DeleteBillingProfileEligibilityResult": {
+      "description": "Eligibility to delete a billing profile result.",
+      "type": "object",
+      "properties": {
+        "eligibilityStatus": {
+          "description": "Status describing if billing profile is eligible to be deleted.",
+          "enum": [
+            "Allowed",
+            "NotAllowed"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "DeleteBillingProfileEligibilityStatus",
+            "modelAsString": true
+          }
+        },
+        "eligibilityDetails": {
+          "description": "Validation details of delete billing profile eligibility.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DeleteBillingProfileEligibilityDetail"
+          },
+          "x-ms-identifiers": []
         }
       }
     },
     "IndirectRelationshipInfo": {
-      "description": "The billing profile details of the partner of the customer for an indirect motion.",
+      "description": "Identifies the billing profile that is linked to another billing profile in indirect purchase motion.",
+      "type": "object",
       "properties": {
         "billingAccountName": {
```

Comment:
Any reason we're projecting names instead of the ARM ids? In general the primary way to reference resources should be through ARM ids.

Reply:
@ramkd12 @amberbhargava to comment here